### PR TITLE
Scaler fixes 3

### PR DESCRIPTION
**Pull request by daschuer**
This includes the vinylemu during direction changed. 
More details can be found here: 
https://github.com/mixxxdj/mixxx/pull/525#issuecomment-84217817

The vinylemu code still suffer a double ramping in some conditions.
Any preferences hoe to fix it? https://bugs.launchpad.net/mixxx/+bug/1435170

### CODE REVIEW

**Comment by ywwg** [comment]
I can see what you're doing -- but I really wish we could write tests for this type of fix

File: src/engine/enginebufferscalelinear.cpp [hunk 1]
```diff
@@ -364,6 +370,16 @@ CSAMPLE* EngineBufferScaleLinear::do_scale(CSAMPLE* buf,
         i += 2 ;
     }
 
+    // Vinylsoundemu:
```
Comment (ywwg) [inline src/engine/enginebufferscalelinear.cpp]: let's not add this yet, since it will make the high-latency problem worse

**Comment by daschuer** [comment]
extra fading loop removed.

File: src/engine/enginebuffer.cpp [hunk 2]
```diff
@@ -896,10 +898,17 @@ void EngineBuffer::process(CSAMPLE* pOutput, const int iBufferSize) {
...
-            if (m_pScale != m_pScaleVinyl) { // linear scaler does this part for us now
-                //XXX: Trying to force RAMAN to read from correct
-                //     playpos when rate changes direction - Albert
-                if (m_speed_old * speed < 0) {
+
+            // Handle direction change.
+            // The linear scaler supports ramping though zero.
+            // This is used for scratching, but not for reverse
+            // For the other, crossfade forward and backward samples
+            if (m_speed_old * speed < 0) {
```
Comment (ywwg) [inline src/engine/enginebuffer.cpp]: just combine these two ifs into a single statement with an &&

**Comment by ywwg** [comment]
Just the one last note I mentioned and then lgtm

**Comment by daschuer** [comment]
Thank you for review.

**Pull request merged**